tests/test_shapes.py

```diff
@@ -136,3 +138,36 @@ def test_bin_import_export():
 
     with raises(Exception):
         Shape.importBin(BytesIO())
+
+
+def test_sample():
+
+    e = ellipse(10, 1)
+    s = segment((0, 0), (1, 0))
+
+    pts1, params1 = e.sample(10)  # equidistant
+    pts2, params2 = e.sample(0.1)  # deflection based
+    pts3, params3 = s.sample(10)  # equidistant, open
+
+    assert len(pts1) == len(params1)
+    assert len(pts1) == 10 - 1  # e is closed
```

Comment:
I see this is not an off by one error but by design (the difference in count for closed/open edges).

I compared with `Sketch`.  `Sketch` has different handling here for closed edges and wires.

```py
import cadquery as cq
from cadquery.func import *
from cadquery.vis import show

# sample 4
wire1 = circle(20)
(pos, params) = wire1.sample(4)
samples = compound([sphere(4).moved(p) for p in pos])

# compare with sketch distribute
sk1 = cq.Sketch().circle(30).edges().distribute(4).circle(2)

show(wire1, samples, sk1)
```

I just want to point out this difference to confirm the design.

Reply:
Good point, I fixed it to be consistent with `Sketch`.